### PR TITLE
feat: create a unique `basctl` server socket for each extension host

### PR DESCRIPTION
**Pull request by alex-gilin**

### CODE REVIEW

File: packages/app-studio-toolkit/src/basctlServer/basctlServer.ts [hunk 1]
```diff
@@ -64,7 +66,9 @@ function createBasctlServer() {
   }
 }
 
-export function startBasctlServer() {
+export function startBasctlServer(context: vscode.ExtensionContext) {
+  // add the socket file name to the environment variables so that the basctl client can use it
```
Comment (veredcon) [inline packages/app-studio-toolkit/src/basctlServer/basctlServer.ts:70]: Maybe add a link to the relevant place where it is used? e.g:
https://github.wdf.sap.corp/devx-wing/bas-cli/blob/master/src/socketClient.js#L4

File: packages/app-studio-toolkit/src/basctlServer/basctlServer.ts [hunk 2]
```diff
@@ -5,8 +5,10 @@ import * as _ from "lodash";
 import { _performAction } from "../actions/performer";
 import { ActionsFactory } from "../actions/actionsFactory";
 import { getLogger } from "../logger/logger";
+import { v4 as uuidv4 } from "uuid";
 
-const SOCKETFILE = "/extbin/basctlSocket";
+// generate a unique socket file name on each run
+const SOCKETFILE = `/extbin/basctl-${uuidv4()}.sock`;
```
Comment (veredcon) [inline packages/app-studio-toolkit/src/basctlServer/basctlServer.ts:11]: So there is one unique socketfile per each app toolkit activation, right? So for multiple windows in each one there will be a different socketfile

**Review by veredcon — APPROVED** [review]

File: packages/app-studio-toolkit/src/basctlServer/basctlServer.ts (hunk 1 above)
Comment (alex-gilin) [reply packages/app-studio-toolkit/src/basctlServer/basctlServer.ts:70]: it is an OSS and the link you mentioned will be unreachable externally

File: packages/app-studio-toolkit/src/basctlServer/basctlServer.ts (hunk 2 above)
Comment (alex-gilin) [reply packages/app-studio-toolkit/src/basctlServer/basctlServer.ts:11]: exactly. unique socket connection for each extension host

**Pull request merged**